Clinical-Genomics/cg
# feat(raredisease) add sex check

## Description
This PR adds a sex check to `cg workflow raredisease metrics-deliver case`, comparing the sex in statusDB against the predicted sex in peddy.

### Added

- Sex check as part of cg workflow raredisease metrics-deliver case

### Changed

-

### Fixed

-


### How to prepare for test

    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b add-raredisease-sex-relatedness-check
    ```

### How to test

- [x] Do `cg workflow raredisease metrics-deliver funkitten`, it should complete successfully
- [x] Change predicted sex from male to female in two samples: `sed -i 's/"predicted_sex_sex_check": "male",/"predicted_sex_sex_check": "female",/g' multiqc/multiqc_data/multiqc_data.json`
- [x] Do `cg workflow raredisease metrics-deliver funkitten`, it should fail.

### Expected test outcome

- [x] Check that code completes as normal:
![image](https://github.com/Clinical-Genomics/cg/assets/57712924/f4bf3e1c-000e-4816-94f4-ef395ea7464d)

- [x] Check that code fails as two males changed to females:
![image](https://github.com/Clinical-Genomics/cg/assets/57712924/83e2c9af-7217-41f4-bb53-e07cde352f1d)


## Review

Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a


## Implementation Plan
